src/screens/Ticketing/Purchase/Overview/components/DurationSelection.tsx

```diff
@@ -0,0 +1,129 @@
+  setSelectedProduct: (product: PreassignedFareProduct) => void;
+};
+
+export default function DurationSelection({
+  color,
+  selectedProduct,
+  setSelectedProduct,
+}: DurationSelectionProps) {
+  const {t, language} = useTranslation();
+  const styles = useStyles();
+
+  const {preassignedFareproducts} = useFirestoreConfiguration();
+
+  const selectableProducts = preassignedFareproducts
+    .filter(productIsSellableInApp)
+    .filter((p) => p.type === selectedProduct.type);
+
+  return (
+    <View style={styles.container}>
+      <ThemeText type="body__secondary" color="secondary">
+        {t(DurationSelectionTexts.title)}
+      </ThemeText>
+      <ScrollView
+        style={styles.durationScrollView}
+        contentContainerStyle={styles.durationContentContainer}
+        horizontal={true}
+        showsHorizontalScrollIndicator={false}
+      >
+        {selectableProducts.map((fp, i) => (
+          <DurationChip
+            color={color}
+            text={getReferenceDataName(fp, language)}
+            selected={selectedProduct == fp}
+            onPress={() => setSelectedProduct(fp)}
+            key={i}
+          />
+        ))}
+      </ScrollView>
+    </View>
+  );
+}
+
+type DurationChipProps = {
+  color: InteractiveColor;
+  text: string;
+  selected: boolean;
+  onPress: () => void;
+};
+
+function DurationChip({color, text, selected, onPress}: DurationChipProps) {
+  const styles = useStyles();
+  const {theme} = useTheme();
+  const {t} = useTranslation();
+
+  const currentColor =
+    theme.interactive[color][selected ? 'active' : 'default'];
+
+  // To make items with and without border the same size, we subtract the border
+  // width from the padding when there is a border.
+  const borderOffset = selected ? theme.border.width.medium : 0;
+
+  return (
+    <TouchableOpacity
+      style={[
+        styles.durationChip,
+        {
+          backgroundColor: currentColor.background,
+          borderColor: theme.interactive[color].outline.background,
+          borderWidth: selected ? theme.border.width.medium : 0,
+          paddingVertical: theme.spacings.medium - borderOffset,
+          paddingHorizontal: theme.spacings.xLarge - borderOffset,
+        },
+      ]}
+      onPress={onPress}
+      accessible={true}
+      accessibilityRole="radio"
+      accessibilityState={{selected}}
+      accessibilityHint={t(DurationSelectionTexts.chipHint)}
+    >
+      <ThemeText
+        style={{
+          color: currentColor.text,
+        }}
+        type={selected ? 'body__secondary--bold' : 'body__secondary'}
+      >
+        {text}
+      </ThemeText>
+    </TouchableOpacity>
+  );
+}
+
+const useStyles = StyleSheet.createThemeHook((theme) => ({
+  container: {
+    marginVertical: theme.spacings.medium,
+  },
+  durationScrollView: {
+    marginTop: theme.spacings.medium,
+    marginBottom: theme.spacings.xLarge,
+    marginLeft: -theme.spacings.medium,
+    marginRight: -theme.spacings.medium,
```

Comment:
This kinda makes a tight coupling between this component and the parent component, as it is dependant on the parent component using the same margin size around this component. Even though this component is not ment to be reused multiple places, we should still try to not create too tight bindings between parent and child. I would rather make the component be "full-width" in the parent, and remove this negative margins.

Reply:
This was also my first thought, but it introduced a lot of extra complexity at the parent level. We would need to split the parent container in three parts, which has an effect on flex functionality as well as padding between components. Also if the component is full width, there still has to be something that handles the horizontal padding of the header ("Select duration"). 

So while I agree that the current implementation isn't a great solution, I think we should stick to handling it within the component. One solution is also to pass the needed overflow margin as a parameter, to make it more universal. Or do you have any other ideas?

Reply:
That's ok. We'll keep it as it is for now.